lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,472 @@
+                license,
+                license_file,
+                readme,
+                repository,
+                homepage,
+                wasmer_extra_flags: None,
+                disable_command_rename: false,
+                rename_commands_to_raw_command_name: false,
+            },
+            dependencies: self.get_dependencies(),
+            command: Self::get_command(&modules, bin_or_lib),
+            module: match bin_or_lib {
+                BinOrLib::Empty => None,
+                _ => Some(modules),
+            },
+            fs: self.get_filesystem_mapping(),
+            base_directory_path: target_file
+                .parent()
+                .map(|o| o.to_path_buf())
+                .unwrap_or_else(|| target_file.clone()),
+        };
+
+        if let Some(parent) = target_file.parent() {
+            let _ = std::fs::create_dir_all(parent);
+        }
+
+        // Test if cargo wapm is installed
+        let cargo_wapm_present = if self.no_cargo_wapm {
+            false
+        } else {
+            let cargo_wapm_stdout = std::process::Command::new("cargo")
+                .arg("wapm")
+                .arg("--version")
+                .output()
+                .map(|s| String::from_utf8_lossy(&s.stdout).to_string())
+                .unwrap_or_default();
+
+            cargo_wapm_stdout.lines().count() == 1
+                && (cargo_wapm_stdout.contains("cargo wapm")
+                    || cargo_wapm_stdout.contains("cargo-wapm"))
+        };
+
+        // if Cargo.toml is present and cargo wapm is installed, add the
+        // generated manifest to the Cargo.toml instead of creating a new wapm.toml
+        let should_add_to_cargo_toml = cargo_toml.is_some() && cargo_wapm_present;
+
+        // If the Cargo.toml is present, but cargo wapm is not installed,
+        // generate a wapm.toml, but notify the user about installing cargo-wapm
+        if !cargo_wapm_present && !self.no_cargo_wapm && cargo_toml.is_some() && !self.quiet {
+            eprintln!(
+                "Note: you seem to have a Cargo.toml file, but you haven't installed `cargo wapm`."
+            );
+            eprintln!("You can build and release Rust projects directly with `cargo wapm publish`: https://crates.io/crates/cargo-wapm");
+            eprintln!("Install it with:");
+            eprintln!();
+            eprintln!("    cargo install cargo-wapm");
+            eprintln!();
+        }
+
+        let note =
+            "# See more keys and definitions at https://docs.wasmer.io/ecosystem/wapm/manifest";
+
+        // generate the wasmer.toml and exit
+        if !should_add_to_cargo_toml {
+            let toml_string = toml::to_string_pretty(&constructed_manifest)?
+                .replace("[dependencies]", &format!("{note}\r\n\r\n[dependencies]"))
+                .lines()
+                .collect::<Vec<_>>()
+                .join("\r\n");
+
+            std::fs::write(target_file, &toml_string)?;
```

Comment:
Let's mention the filename when a write fails.

```suggestion
            std::fs::write(&target_file, &toml_string)
                .with_context(|| format!("Unable to write to \"{}\"", target_file.display()))?;
```